tests/run_tests.py

```diff
@@ -351,33 +399,56 @@ def run_system(config_file, vic_exe, test_data_dir, out_dir, driver):
         error_message = ''
 
         try:
-            # Run the VIC simulation
-            returncode = vic_exe.run(test_global_file, logdir=dirs['logs'],
-                                     **run_kwargs)
-            test_complete = True
+            if 'restart' in test_dict:
+                for j, test_global_file in enumerate(list_test_global_file):
+                    returncode = vic_exe.run(test_global_file,
+                                             logdir=dirs['logs'])
+                    # Check return code
+                    check_returncode(returncode,
+                                     test_dict.pop('expected_retval', 0))
+            else:
+                returncode = vic_exe.run(test_global_file, logdir=dirs['logs'],
+                                         **run_kwargs)
+                # Check return code
+                check_returncode(returncode,
+                                 test_dict.pop('expected_retval', 0))
 
-            # Check return code
-            check_returncode(returncode, test_dict.pop('expected_retval', 0))
+            test_complete = True
 
             # check output files (different tests depending on driver)
             if test_dict['check']:
-                fnames = glob.glob(os.path.join(dirs['results'], '*'))
 
                 # Check that the simulation completed for all grid cells
-                if 'complete' in test_dict['check'] and driver == 'classic':
-                    test_classic_driver_all_complete(fnames)
+                if 'complete' in test_dict['check']:
+                    fnames = glob.glob(os.path.join(dirs['results'], '*'))
+                    if driver == 'classic':
+                        test_classic_driver_all_complete(fnames)
```

Comment:
- For putting the generation of the fnames into the individual checks - I thought this is only needed for some (but not all) tests, so it makes more sense to only do it when needed. But yeah if there are multiple classic driver tests, this step would be done for multiple times with my changes (although I don't think this would take long).
- For putting `if driver == 'classic'` inside - I guess I thought there might be a 'complete' check for image driver at some point?